src/db/mod.rs

```diff
@@ -147,6 +147,8 @@ pub trait Db: Send + Debug {
 
     fn box_clone(&self) -> Box<dyn Db>;
 
+    fn check(&self) -> Result<bool, DbError>;
```

Comment:
shouldn't we go w/ async here like other db methods?

Reply:
probably, although not really looking forward to juggling either states.